serde/src/java/org/apache/hadoop/hive/serde2/json/HiveJsonReader.java

```diff
@@ -466,6 +476,16 @@ private byte[] getByteValue(final JsonNode binaryNode) throws SerDeException {
     }
   }
 
+  private BinaryEncoding getBinaryEncodingForNode(JsonNode binaryNode) {
+    String jsonValue = binaryNode.textValue();
+
+    if(jsonValue == null || jsonValue.length() % 4 != 0 || !BASE64_PATTERN.matcher(jsonValue).matches()) {
+      return BinaryEncoding.RAWSTRING;
+    }
+
+    return BinaryEncoding.BASE64;
```

Comment:
return `binaryEncoding`?

Reply:
Thank you so for the review! 
Dropped this variable from HiveJsonReader as here we are detecting the encoding automatically. Although, HiveJsonWriter will use BASE64 encoding to serialized data for insert statements.

Reply:
@deniskuzZ  `HiveJsonWriter` is used for serializing values to JSON during `INSERT` statements, so using a single encoding type should suffice. If future bugs arise (though no such scenarios are currently identified), we can consider supporting both encoding types as needed.

Reply:
wait, why do we need to determine the encoding if we could simply pass the encoding used in the writer as it was done originally?

Reply:
The encoding needs to be determined because loading JSON directly (e.g., using `LOAD DATA INPATH`) causes issues. Users must encode their JSON to BASE64 beforehand; otherwise, an error is thrown. This behavior is inconsistent since we previously supported deserializing RAWSTRING JSON values provided by users.

Steps to reproduce this issue and the inconsistency are detailed in the JIRA description: [HIVE-28632](https://issues.apache.org/jira/browse/HIVE-28632).

Reply:
yes, I got that. updated the comment

Reply:
My bad, I just saw the updated comment. You are right. 
Thank you so much for the review again!